src/Runner.Worker/ActionManager.cs

```diff
@@ -1004,7 +1004,7 @@ private ActionSetupInfo PrepareRepositoryActionAsync(IExecutionContext execution
                 if (actionDefinitionData.Execution.ExecutionType == ActionExecutionType.Container)
                 {
                     var containerAction = actionDefinitionData.Execution as ContainerActionExecutionData;
-                    if (containerAction.Image.EndsWith("Dockerfile") || containerAction.Image.EndsWith("dockerfile"))
+                    if (DockerUtil.IsDockerfile(containerAction.Image))
```

Comment:
Can we update the error on line 1029

Reply:
should we swap this if condition to be safer?

So do

```
if (containerAction.Image.StartsWith("docker://" ...)
{

} else if(DockerUtil.IsDockerfile()
{

}
else
{
   throw
}

```

Reply:
I think that we will break some of the workflows by doing that if someone specified `image: ubuntu:latest`. Right now, the logic tries to see if we start with `docker://`, and if not, we try to parse it as a Dockerfile. If Dockerfile is not provided, we are falling back to Docker Hub pull image. If we do this, someone specifying `image: ubuntu:latest` and not docker:// is going to have their workflow throwing exceptions. I checked if I define it as `image: ubuntu:latest`, inside ContainerActionHandler, we would get `ubuntu:latest`

Reply:
Don't we just [throw](https://github.com/actions/runner/pull/1738/files#diff-5b282ef924f268139b4015eb491825d3e35bfa6cbe46962e3bef3bd8fa05dfccL1029) if it doesn't start with `docker://`

Do we catch that throw somewhere or something?

Reply:
I am only worried that some stuff in [docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-running-a-job-within-a-container) are explaining it without the `docker://`
Usually, you specify docker:// when you want to talk to the Docker socket, and if you want to specify an online registry, you would have to do it like `docker://gcr.io/nikola-jokic/my-immage` let's say

Reply:
```
if (containerAction.Image.StartsWith("docker://" ...)
{

} else if(DockerUtil.IsDockerfile()
{

}
else
{
   throw
}
```

This above would throw on the config below wouldn't it?

```
container:
      image: node:14.16
```